### PR TITLE
style : Hide subtitle if it is empty

### LINKED ISSUE #1579: Hide subtitle if it is empty
**Actual Behaviour**
Even if the subtitle is empty, it is still displayed and consumes unnecessary space which makes the card look ugly 
image

**Expected Behaviour**
It shouldnt be displayed if it is empty so that the card looks crisp
image

**Steps to reproduce it**
Visit any track in the current app

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #1579

Changes: Add ->if(session.getSubtitle().isEmpty())
            holder.sessionSubtitle.setVisibility(View.GONE);

in SessionsListAdapter.
And add bottom margin of 8dp to titleLinearLayout in tracksactivity_item.xml

Screenshots for the change: 
image

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 1]
```diff
@@ -110,6 +110,7 @@ public SessionsListAdapter(Context context, List<Session> sessions,int type) {
     }
 
     public void setTrackName(String trackName) {
+
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Please don't add extra lines while creating PRs

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 2]
```diff
@@ -214,6 +218,7 @@ public void accept(@NonNull String speakerList) throws Exception {
             case trackWiseSessionList :
                 holder.trackImageIcon.setVisibility(View.GONE);
                 holder.sessionTrack.setVisibility(View.GONE);
+                System.out.println("HIIII");
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Please remove

File: android/app/src/main/res/layout/tracksactvity_item.xml [hunk 3]
```diff
@@ -24,7 +24,8 @@
                 android:layout_height="wrap_content"
                 android:background="@color/color_primary"
                 android:orientation="horizontal"
-                android:padding="@dimen/padding_medium">
+                android:padding="@dimen/padding_medium"
+                android:layout_marginBottom="8dp">
```
Comment (iamareebjamal) [inline android/app/src/main/res/layout/tracksactvity_item.xml]: Is any existing `dimens` value usable?

If so, please use it

**Comment by iamareebjamal** [comment]
Also, please squash your commits

**Comment by iamareebjamal** [comment]
Please squash the commits and we can merge it

**Comment by mayank8318** [comment]
Squashed!
Really sorry for my carelessness

**Comment by Shailesh351** [comment]
@mayank8318  You haven't squashed commits yet. check it

**Comment by iamareebjamal** [comment]
If you are facing some difficulties in squashing, we can help

**Comment by mayank8318** [comment]
Yes I am new to git and could definitely use some help.
I used -> git rebase -i HEAD~5 to squash. Is this not the right way to do it?
Thanks :)

**Comment by iamareebjamal** [comment]
Use `git rebase -i HEAD~5`

Then to squash all commits after first, let the first one remain pick, 

And change all subsequent ones to be squash, like 

```
pick f87uey65
squash iu8976g
squash tre34s2b
squash tr22nshg
squash oiu765qj
```

Then save it, a new prompt will open with several commit messages, delete all except first and save
Now, you have to force push it to your branch as the history has changed

`git push -f origin Subtitle`

**Comment by mayank8318** [comment]
I did all that except delete the other commit messages and force pushing it. I did a normal push.
So what should I do next?
Thanks!

**Comment by iamareebjamal** [comment]
Rebase it again, if you see the 5 commits again, do what I told in previous comments

If you don't see the 5 commits, just one, then just force push

**Comment by mayank8318** [comment]
I am getting this if i do git rebase -i HEAD~5
pick dee59ed fix : Displaying same time for session when timezone is $
pick f0d9893 feat: Save splash logo during apk generation (#1556)
pick 8bfc6b5 Push docker images to public docker repo
pick 3cb5996 style : Hide subtitle if it is empty Fixes #1579
pick 1058114 style : Hide subtitle if it is empty Fixes #1579
pick bab7617 Remove unnecesarry print statements and use value from d$
pick c25ab96 feat: Replace callbacks to lambda in remaining parts (#1$
pick 07a9261 fix: add sort option for org and country only when neede$
pick c0d5944 OSX docs for generic installation of app generator (#157$
pick 92b0b9f fix: getting event info from the assets folder (#1562)
pick e77b485 Remove unnecessary empty line(112)
pick 3895c0d Delete hs_err_pid2719.log

**Comment by iamareebjamal** [comment]
pick dee59ed fix : Displaying same time for session when timezone is $
pick f0d9893 feat: Save splash logo during apk generation (#1556)
pick 8bfc6b5 Push docker images to public docker repo
pick c25ab96 feat: Replace callbacks to lambda in remaining parts (#1$
pick 07a9261 fix: add sort option for org and country only when neede$
pick c0d5944 OSX docs for generic installation of app generator (#157$
pick 92b0b9f fix: getting event info from the assets folder (#1562)
pick 3cb5996 style : Hide subtitle if it is empty Fixes #1579
squash 1058114 style : Hide subtitle if it is empty Fixes #1579
squash bab7617 Remove unnecesarry print statements and use value from d$
squash e77b485 Remove unnecessary empty line(112)
squash 3895c0d Delete hs_err_pid2719.log

**Comment by iamareebjamal** [comment]
This happens when you merge the development branch into yours instead of rebasing.

Always rebase instead of merging

Now just use the ordering I gave above to squash your commits

**Comment by mayank8318** [comment]
error: could not apply 1058114... style : Hide subtitle if it is empty

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

Could not apply 1058114e8299aaa458f1cbcacafaad89bbbefbf0... style : Hide subtitle if it is empty Fixes #1579

I get this error?

**Comment by iamareebjamal** [comment]
There are conflicts while rebasing, please solve them

**Comment by iamareebjamal** [comment]
Awesome, thanks :+1:

**Comment by mayank8318** [comment]
Thanks a lot for helping me with this :)

**Pull request merged**